src/server/saved_objects/service/lib/repository_provider.js

```diff
@@ -53,6 +55,7 @@ export class SavedObjectsRepositoryProvider {
       schema: this._schema,
       serializer: this._serializer,
       onBeforeWrite: this._onBeforeWrite,
+      allowedTypes: this._allowedTypes.concat(extraTypes),
```

Comment:
similar issue here, as above: this could lead to duplicate entries in the `allowedTypes` property, or types that don't actually exist.